apps/prairielearn/src/middlewares/autoEnroll.ts

```diff
@@ -0,0 +1,32 @@
+import asyncHandler = require('express-async-handler');
+import { idsEqual } from '../lib/id';
+import { insertCheckedEnrollment } from '../models/enrollment';
+
+export default asyncHandler(async (req, res, next) => {
+  // If the user does not currently have access to the course, but could if
+  // they were enrolled, automatically enroll them. However, we will not
+  // attempt to enroll them if they are an instructor (that is, if they have
+  // a specific role in the course or course instance) or if they are
+  // impersonating another user.
+  if (
+    idsEqual(res.locals.user.user_id, res.locals.authn_user.user_id) &&
```

Comment:
This conditional is new relative to the original home. I added it to hopefully keep the behavior consistent, as the enrollment was previously added before any overrides were taken into account, but now it runs after those have been resolved. Ensuring the IDs are equal is my way of checking that the user is not impersonating another.